### ISSUE TITLE
Auto Close OSD broken on Kodi 19.2

### ISSUE THREAD

**Issue by JFG1000** [SerpentDrago/skin.auramod#207]
Hi there I've noticed after some testing this morning, since updating to 19.2 on my nvidia shield pro, the auto Close OSD is broken, we now have to manually press the back button to close it during video playback, could this be looked into? Many thanks

Kodi 19.2
Auramod latest version 
Nvidia Shield Pro

**Comment by Dealz-416LS** [comment]
Experiencing the same issue, `Auto close OSD controls after delay (in seconds)` is no longer working.

**Comment by JFG1000** [comment]
Not sure this skin is being worked on anymore to be honest.

**Comment by SerpentDrago** [comment]
Sorry no , its no longer being worked on . 

Jurial recently did a custom solution to add this feature natively his skin Eminence 2 Matrix .  (not using skinhelper or any other deps ) 

Your welcome to add that solution . 

https://github.com/jurialmunkey/skin.eminence.2/commit/6b5b74fd0fe348b9e69eee4f0ae2801bb8a79a60

Good luck !

**Comment by JFG1000** [comment]
Thanks! For me personally I'm sticking with Auramod for as long as possible, I can live with pressing the back button to close OSD, it's no biggie

**Comment by SerpentDrago** [comment]
> Thanks! For me personally I'm sticking with Auramod for as long as possible, I can live with pressing the back button to close OSD, it's no biggie

my point is you can take the code and modify auramod to do the same thing . its not hard . and wouldn't take long , i just have 0 desire to

**Comment by JFG1000** [comment]
Would anyone be willing to fix this one issue? I'd happily pay them at this rate or at least show me how to do it myself...

**Comment by Goldenfreddy0703** [comment]
I can probably get the person who works with the cosmic skin to help with auramod on this future but I'm not sure if he is able to work on a different skin or maybe drinfernoo help us.

**Comment by Goldenfreddy0703** [comment]
Or if I can know which lines to edit in auramod, I can probably do it but I'm not that good at coding.

**Comment by JFG1000** [comment]
Mate if you could see what you can sort out that be amazing apparently it's pretty simple to edit the code according to @SerpentDrago but I have no idea

**Comment by inb4after** [comment]
@JFG1000 @Dealz-416LS @Goldenfreddy0703 

EDIT: To fix this issue, update `script.skin.helper.service` from Marcelveldt BETA repo via AuraMOD AIO repo (File manager source: `https://serpentdrago.github.io/`).


~~I've updated to 19.3 and I cannot reproduce the issue, try testing this fix:~~

~~1. `Settings > Skin Settings > Video / Music OSD > Video OSD > Auto close OSD controls after delay (in seconds)`.
2. Set a `blank` value using `Backspace` and press `Done`.
2. Play a video.
3. Confirm that OSD does not close automatically.
4. While the video is playing in the background go back to `Settings > Skin Settings > Video / Music OSD > Video OSD > Auto close OSD controls after delay (in seconds)`.
5. Set the desired value and press `Done`.
5. Return to the video and test if OSD now closes automatically.~~

**Comment by JFG1000** [comment]
Will try this after work and keep you posted. If there's no luck would you be able to fix this for us? Would he massively appreciated

**Comment by inb4after** [comment]
> Will try this after work and keep you posted. If there's no luck would you be able to fix this for us? Would he massively appreciated

If I find some free time I will look into it further, but hopefully this fix works!

**Comment by JFG1000** [comment]
Absolute legend I'll message back here tonight mate

**Comment by JFG1000** [comment]
What I do know is on a fire box it isn't broken on 19.3 only on android I've noticed

**Comment by inb4after** [comment]
Tested on Shield and I have reproduced the issue. 

The auto-close OSD is handled by a `script.skin.helper.service` function. Despite the version being the same on PC and Shield (`1.1.33`), this function doesn't work only on the Shield (and other android devices).

I downloaded and installed the latest version (`1.1.40`) of the above addon from Marvelveldt BETA repository, via AuraMOD AIO repo (File manager source: `https://serpentdrago.github.io/`) and the issue is now fixed.

It seems the version we installed previously for AuraMOD setup was orphaned after Leia and was only supposed to be used with Leia; the Matrix fixes for this addon were only implemented in `1.1.34`.

**Comment by JFG1000** [comment]
Legend thanks mate I'll give this a try tonight, is there an official repo do you know we can install so it gets future updates?

**Comment by SerpentDrago** [comment]
> Tested on Shield and I have reproduced the issue.
> 
> The auto-close OSD is handled by a `script.skin.helper.service` function. Despite the version being the same on PC and Shield (`1.1.33`), this function doesn't work only on the Shield (and other android devices).
> 
> I downloaded and installed the latest version (`1.1.40`) of the above addon from here: https://github.com/kodi-community-addons/script.skin.helper.service and the issue is now fixed.
> 
> It seems the version we installed previously for AuraMOD setup was orphaned after Leia and was only supposed to be used with Leia; the Matrix fixes for this addon were only implemented in `1.1.34`.

The  Version from Marcels Repo is 1.1.40 > https://github.com/kodi-community-addons/repository.marcelveldt/tree/master/matrix/script.skin.helper.service

Which is exactly where people should be getting it , PER the Instructions for installing auramod > Step 3 > https://github.com/SerpentDrago/skin.auramod/wiki/Install-Instructions

So i have no idea why someone would not be on the latest version . if they followed the install instructions !!!

**Comment by JFG1000** [comment]
Pretty sure I followed all of these instructions already as I did a fresh install of aura I will double check later

**Comment by SerpentDrago** [comment]
you can simply install from Marcels repo , it should be installed per My AuraMod matrix install instructions  and SHOULD update from there as proved here > https://raw.githubusercontent.com/kodi-community-addons/repository.marcelveldt/master/matrix/addons.xml
![image](https://user-images.githubusercontent.com/21133858/145434050-52dae7a2-365d-4729-8b9b-a792bf393768.png)

That Step is always been important other wise you will use outdated versions from the official kodi repo 


as you can see Marcels repo contains the latest version . AND Aura AIO repo contains the latest marcels repo  zip as proved here> https://github.com/SerpentDrago/repository.auramod.aio/blob/main/repo/matrix/repository.marcelveldt/addon.xml

**Comment by Dealz-416LS** [comment]
I just checked and my version of Skin Helper Service is 1.1.40 but I also had this issue on 19.2 where the OSD would no longer automatically close after x seconds.

I am not sure which version I had at the time.

Update: 
My mistake, I actually was on 1.1.33 not sure how, I've done installs on AuraMod with the new installation step and the old steps before that. Anyway I updated to 1.1.40 and OSD still doesn't work.
